app/main/views/questionnaire.py

```diff
@@ -115,26 +135,33 @@ def redirect_to_questionnaire_page(eq_id, form_type, collection_id, location):
                             location=location))
 
 
-def same_survey(eq_id, form_type, collection_id):
+def _same_survey(eq_id, form_type, collection_id):
     metadata = get_metadata(current_user)
     current_survey = eq_id + form_type + collection_id
     metadata_survey = metadata["eq_id"] + metadata["form_type"] + metadata["collection_exercise_sid"]
     return current_survey == metadata_survey
 
 
-def render_page(location, is_valid):
-    context = g.questionnaire_manager.get_rendering_context(location, is_valid)
-    template = TemplateRegistry.get_template_name(location)
-    navigator = g.questionnaire_manager.navigator
-    previous_location = navigator.get_previous_location(get_answers(current_user), location)
-    return render_template(template, context, previous_location)
-
-
-def render_template(template, context, previous_location):
+def _render_template(location, context, rendered_schema_json=None, page=None):
```

Comment:
- I think the `page` parameter should be renamed `template` and explicitly pass in the filename 'questionnaire.html'
- `rendered_schema_json` doesn't appear to be passed in by anything